packages/jazz-tools/src/tools/coValues/inbox.ts

```diff
@@ -115,135 +222,124 @@ export class Inbox {
       message: InstanceOfSchema<M>,
       senderAccountID: ID<Account>,
     ) => Promise<O | undefined | void>,
-    options: { retries?: number } = {},
+    options?: { concurrencyLimit?: number },
   ) {
     const processed = new Set<`${SessionID}/${number}`>();
     const failed = new Map<`${SessionID}/${number}`, string[]>();
     const node = this.account.$jazz.localNode;
 
-    this.processed.subscribe((stream) => {
-      for (const items of Object.values(stream.items)) {
-        for (const item of items) {
-          processed.add(item.value as TxKey);
-        }
+    // Create queue instance inside subscribe function
+    const concurrencyLimit = options?.concurrencyLimit ?? 10;
+    const messageQueue = new MessageQueue(concurrencyLimit);
+
+    const processedFeed = new IncreamentalFeed(this.processed);
+
+    // Track the already processed messages, triggered immediately so we know the messages processed in the previous sessions
+    this.processed.subscribe(() => {
+      for (const { changes } of processedFeed.getNewItems()) {
+        processed.add(changes[0] as TxKey);
       }
     });
 
     const { account } = this;
-    const { retries = 3 } = options;
 
-    let failTimer: ReturnType<typeof setTimeout> | number | undefined =
-      undefined;
+    const messagesFeed = new IncreamentalFeed(this.messages);
 
-    const clearFailTimer = () => {
-      clearTimeout(failTimer);
-      failTimer = undefined;
-    };
+    // Set up the message processing handler for the queue
+    messageQueue.setProcessMessageHandler(async (txKey, messageId) => {
```

Comment:
Nit: why not make processMessage and handleError functions that are provided when creating MessageQueue?